src/language-js/printer-estree.js

```diff
@@ -5077,6 +5079,37 @@ function printMemberChain(path, options, print) {
   ]);
 }
 
+/**
+ * @param {import('estree').Node} node
+ * @returns {boolean}
+ */
+function isSimple(node) {
+  if (
+    !node ||
+    node.type === "Literal" ||
+    node.type === "BooleanLiteral" ||
+    node.type === "NullLiteral" ||
+    node.type === "NumericLiteral" ||
+    node.type === "StringLiteral" ||
+    node.type === "Identifier"
```

Comment:
`ThisExpression`, `Super`, `BigIntLiteral`, `PrivateName`, `ArgumentPlaceholder`

Reply:
Will add all these, plus some test inputs to make sure they a look sensible.